gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -285,6 +286,22 @@ internal open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
         "implementations."
     }
   }
+
+  private val variantCache = mutableMapOf<String, Variant>()
```

Comment:
nit: move this to the top with other `val`s?

Reply:
Also you might be able to just make the map itself synchronized:

```kotlin
private val variantCache = ConcurrentHashMap<String, Variant>()
```

and then can drop the `@Synchronized` method below?

Reply:
I made it specifically so to ensure that the filter action is only executed once. `getOrPut` on a `ConcurrentHashMap` is not threadsafe. The documentation calls this out as well:
```
Note that the operation is not guaranteed to be atomic if the map is being modified concurrently.
```

Reply:
Ah sorry, forgot to mention that at that point you would also use `computeIfAbsent` instead of `getOrPut`, which is a method of `ConcurrentHashMap` and not an extension.

Reply:
Ah, right. That works!